content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -6,12 +6,19 @@ const {GroupedActivityFeed} = require("components/ActivityFeed/ActivityFeed");
 const Spotlight = require("components/Spotlight/Spotlight");
 const Search = require("components/Search/Search");
 const {actions} = require("actions/action-manager");
+let REPORT_RENDER_EVENT = true;
 
 const NewTabPage = React.createClass({
   // TODO: Replace with real search api via addon
   onSearch(value) {
     this.props.dispatch(actions.NotifyPerformSearch(value));
   },
+  componentWillUpdate(nextProps, nextState) {
+    if (nextProps.isReady && REPORT_RENDER_EVENT) {
+      this.props.dispatch(actions.NotifyTelemetry("NEWTAB_RENDER"));
```

Comment:
So, there's something funny going on with these notifications.  I tested at a bit by placing window.confirm() right after the notification is sent, and sure enough - only half of the page rendered.  Recent history appears after I click confirm button. And yes, we are going through this function multiple times on first render, hence the question why?  Will bug @k88hudson on Monday